### PR TITLE
core/freezer - retry sync write to disk

### PR DESCRIPTION
**Pull request by antonydenyer**

### CODE REVIEW

File: core/rawdb/freezer.go [hunk 1]
```diff
@@ -259,17 +259,28 @@ func (f *freezer) TruncateAncients(items uint64) error {
...
-	if errs != nil {
+	if errs != nil && retry < 10 {
 		return fmt.Errorf("%v", errs)
+	} else if errs != nil {
+		log.Info("RETRY %i %v", retry, errs)
+		return f.SyncRetry(retry+1)
 	}
 	return nil
 }
+// Quorum
```
Comment (baptiste-b-pegasys) [inline core/rawdb/freezer.go]:
```suggestion
// End Quorum
```

File: core/rawdb/freezer.go [hunk 2]
```diff
@@ -259,17 +259,28 @@ func (f *freezer) TruncateAncients(items uint64) error {
...
 		if err := table.Sync(); err != nil {
 			errs = append(errs, err)
 		}
 	}
-	if errs != nil {
+	if errs != nil && retry < 10 {
 		return fmt.Errorf("%v", errs)
+	} else if errs != nil {
+		log.Info("RETRY %i %v", retry, errs)
+		return f.SyncRetry(retry+1)
```
Comment (baptiste-b-pegasys) [inline core/rawdb/freezer.go]:
```suggestion
	hasError := len(errs) > 0
	if hasError && retry < 10 {
		log.Info("sync", "retry", retry, "errors", errs)
		return f.SyncRetry(retry+1)
	} else if hasError {
		return fmt.Errorf("%v", errs)
```

**Review by baptiste-b-pegasys — CHANGES_REQUESTED** [review]

File: core/rawdb/freezer.go [hunk 3]
```diff
@@ -259,17 +259,29 @@ func (f *freezer) TruncateAncients(items uint64) error {
...
 	for _, table := range f.tables {
 		if err := table.Sync(); err != nil {
 			errs = append(errs, err)
 		}
 	}
-	if errs != nil {
+	hasError := len(errs) > 0
+	if hasError && retry < 10 {
+		log.Info("sync", "retry", retry, "errors", errs)
+		return f.SyncRetry(retry+1)
```
Comment (baptiste-b-pegasys) [inline core/rawdb/freezer.go]: should we add some (exponential backoff) delay?

Comment (antonydenyer) [reply core/rawdb/freezer.go]: yeah probably not a bad idea - or even just a little :)

**Review by baptiste-b-pegasys — APPROVED** [review]

**Pull request merged**